### ISSUE TITLE
Project abandoned by originator?

### ISSUE THREAD

**Issue by oyvindlr** [chriskiehl/Gooey#913]
The latest release was three years ago, and the latest commit was two years ago. Meanwhile, there are a multitude of issues and pull requests that go unnoticed. Presumably @chriskiehl has abandoned this project.  However, the project seems to be popular, and there are many potential contributors. 
So my question is, could you, @chriskiehl  let someone else take control of the project? Alternatively, if he does not answer, does anyone want to start a semi-official fork?

**Comment by AndrewBuiltThis** [comment]
Agreed - would love if this got some small QoL updates and had some ongoing support but don't want to usurp the OG

**Comment by chriskiehl** [comment]
Howdy folks! 

Is this project abandoned? In my head, no. I'd say more 'on pause' while I pursue other things (I'm currently writing a book!). This little project has been a big part of my programming life (it stretches all the way back to when I was taking my very first steps with software development), so it'd be very hard for me to totally abandon it. However, it's hard to argue with @oyvindlr. Three years between releases is a crummy state for the project to be in.   

I'm happy to give commit rights to a few folks. I'd like to hear where you want to take the project first. 

Despite my absence, I have strong opinions on what Gooey should definitely _not_ be. Not every PR warrants a merge. Not every pet feature request needs implemented. Gooey is an extremely simple tool and should remain that way. It doesn't need to do _everything_ people ask for (you'll need to gently remind them that terminals already exist). Its sights are ultimately very modest. 

[I highly favor conservatism over "shipping features"](https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9). Correctness and stability trump velocity. If my flavor of stodgy crankiness aligns with yours, commit access granted 😄

**Comment by oyvindlr** [comment]
Thank you very much for the update, @chriskiehl ! Whoever gets commit rights should obviously be someone familiar with the code, and unfortunately that is not me at the moment. I was looking through the pull requests to see if anyone had any major contributions that might suggest they were familiar with more than a small piece of the code, but unfortunately I couldn't find anyone fitting (I may have overlooked someone though). If anyone would like to volunteer, please speak up! How about @AndrewBuiltThis ?

**Comment by sbillinge** [comment]
Did this get resolved?  I also would love it if this could be, at least minimally, maintained.  Even if it is just making it run on latest python and pushing to PyPi and conda-forge.  @chriskiehl  I hope the book is going well!

**Comment by oyvindlr** [comment]
Still waiting for someone to volunteer to take over 😊

lør. 16. nov. 2024, 04:45 skrev Simon Billinge ***@***.***>:

**Comment by sbillinge** [comment]
I am willing to volunteer on the understanding that the immediate goal is minimal updates that demonstrably improve current functionality, fix bugs, and keep it operating with current releases of python and its dependencies.....  For the record, we (myself and my research group) maintain the [diffpy stack](https://github.com/diffpy) of software for diffraction and PDF analysis for the chemistry/materials communities.  We release on Pypi and conda-forge.

**Comment by oyvindlr** [comment]
That sounds great, and it seems you have the necessary credentials! Hopefully, @chriskiehl sees this and is willing to give you commit rights. 
By the way, it seems like his book is coming along great! [https://mng.bz/BgQv]

**Comment by sbillinge** [comment]
> I am willing to volunteer on the understanding that the immediate goal is minimal updates that demonstrably improve current functionality, fix bugs, and keep it operating with current releases of python and its dependencies..... For the record, we (myself and my research group) maintain the [diffpy stack](https://github.com/diffpy) of software for diffraction and PDF analysis for the chemistry/materials communities. We release on Pypi and conda-forge.

@chriskiehl any thoughts on this? I would love to move this forward.

**Comment by chriskiehl** [comment]
@sbillinge invite sent!

**Comment by sbillinge** [comment]
thank you!

**Comment by plasorak** [comment]
Any news on this? I see the last change was over 3 years ago, which makes me think that this project isn't maintained, that's a shame because it's great!

**Comment by sbillinge** [comment]
@chriskiehl is busy with his work and writing a book.  I will be helping with releases, but also unexpectedly got busy this summer.  I expect to be able to do a release very shortly though as we are just about to push out a big refactor of our diffpy stack.  I hope we can get to regular releases again shortly.
